### ISSUE TITLE
A better approach to check if we need to download the data

### ISSUE THREAD

**Issue by agila5** [ropensci/osmextract#119]
See here: https://twitter.com/jakub_nowosad/status/1295030990121979908

**Comment by Robinlovelace** [comment]
The [`pins` package](https://pins.rstudio.com/) seems good but there are loads of options. What advantages were you thinking of from a user perspective? Benefit of our `file.exists()` approach from a developer perspective: simplicity.

**Comment by agila5** [comment]
> What advantages were you thinking of from a user perspective

The main thing that I don't like with the current approach (but it's not a big deal I think) is that the user cannot modify the names of the files that are stored in the download_directory otherwise all the "assumptions" behind the oe_* functions are wrong. I know that this is not an important drawback (since I cannot think of any good reason to modify the names of those files). 

The most important part was creating a "self-note" to study the principles behind caching (sooner or later), even if these ideas are not implemented here. 

Maybe also related: https://github.com/ropensci/osmdata/issues/159

**Comment by Robinlovelace** [comment]
Makes sense, there could be advantages of using a caching framework/package rather than the simple approach we have and good to think about it. I think this is a discussion issue rather than a 'to do' issue and will mark it as such, if you think it's worth doing it would also get thumbs up from me if you think it's worth the extra dependency/code complexity.
